src/Sentry.Hangfire/SentryServerFilter.cs

```diff
@@ -0,0 +1,57 @@
+using Hangfire.Server;
+using Sentry.Extensibility;
+
+namespace Sentry.Hangfire;
+
+internal class SentryServerFilter : IServerFilter
+{
+    internal const string SentryMonitorSlugKey = "SentryMonitorSlug";
+    internal const string SentryCheckInIdKey = "SentryCheckInIdKey";
+
+    private readonly IHub _hub;
+    private readonly IDiagnosticLogger? _logger;
+
+    public SentryServerFilter() : this(HubAdapter.Instance)
+    { }
+
+    internal SentryServerFilter(IHub hub, IDiagnosticLogger? logger = null)
+    {
+        _hub = hub;
+        _logger = logger ?? _hub.GetSentryOptions()?.DiagnosticLogger;
+    }
+
+    public void OnPerforming(PerformingContext context)
+    {
+        var monitorSlug = context.GetJobParameter<string>(SentryMonitorSlugKey);
+        if(monitorSlug is null)
+        {
+            var jobType = context.BackgroundJob.Job.Type;
+            var jobMethod = context.BackgroundJob.Job.Method;
+            _logger?.LogWarning("Skipping creating a check-in for '{0}.{1}'. " +
+                                "Failed to find Monitor Slug the job. You can set the monitor slug " +
+                                "by setting the 'SentryMonitorSlug' attribute.", jobType, jobMethod);
+            return;
+        }
+
+        var checkInId = _hub.CaptureCheckIn(new SentryCheckIn(monitorSlug, CheckInStatus.InProgress));
```

Comment:
APIs where we need to do `method(new Something()` feel janky, can we have an overload that takes the two arguments directly?

Reply:
this was resolved but it's still doing `new CheckIn` ?